### ISSUE TITLE
iOS Application freezes when presenting a Survey (DEBUG)

### ISSUE THREAD

**Issue by mohshin-shah** [UserLeap/userleap-ios-sdk-releases#90]
Tech Stack
Platform: iOS Swift
Version: 4.21.0
Usage: SPM
We are using a V2 method as below:

```
func tracking(from vc: UIViewController?) {
    let eventPayload: EventPayload = .init(
      eventName: "someEventName",
      properties: someProps
    ) { [weak vc] state in
      guard let viewController = vc else { return }
      DispatchQueue.global(qos: .userInitiated).async {
        myPresentSurvey(from: viewController, state: state)
      }
    }
    
    UserLeapKit.Sprig.shared.track(payload: eventPayload)
}

private func myPresentSurvey(from vc: UIViewController, state: SurveyState) {
    guard case .ready = state else { return }
    DispatchQueue.main.async {
      UserLeapKit.Sprig.shared.presentSurvey(from: vc)
    }
  }

```
Please let us know if we are missing any usage instruction or there is some recent issue in the SDK itself

**Comment by tiagojsalmeida** [comment]
We are seeing a similar issue with version 4.16.

**Comment by helloandrewpark** [comment]
@mohshin-shah there is no 4.21.0 available. could you double check your version?

**Comment by helloandrewpark** [comment]
@tiagojsalmeida we're unable to reproduce this on 4.16.0, could you share a sample project that reproduces the issue?

**Comment by gongstr** [comment]
@tiagojsalmeida after some more testing, we were able to identify this issue. Looks like this issue was introduced since 4.15.0. We'll release an update 4.16.1 coming up early next week.

**Comment by tiagojsalmeida** [comment]
@gongstr any updates on this 4.16.1 release?

**Comment by helloandrewpark** [comment]
we have just released 4.17.0. please give that a shot!
